### PR TITLE
Compat with new buildcraft fuel

### PR DESCRIPTION
**Pull request by jasperfirecai2**
**The Issue**
Related to Issue #1861 
 
**The Proposal**
Adds the new buildcraft oils as fuels for railcraft, primarily for steam boiler
Based on the following:
```
Base fuel values in Burn time, MJ/t, total MJ produced in combustion engine
Oil: 500s, 3/t, 30k total
Heavy oil: 2000s, 2/t, 80k total
Dense oil: 1500s, 4/t, 120k total
Distilled oil: 1875s, 1/t, 37.5k total
Dense fuel: 4500s, 4/t, 360k total
Mixed heavy fuels: 960s, 5/t, 96k total
Light fuel: 750s, 6/t, 90k total
Mixed light fuels: 500s, 3/t, 30k total
Gaseous fuel: 93s, 8/t, 15k total
```
Burn time ratio Based on burn times above with baseline where Light fuel's burn time = old 'Fuel' burn time

 
**Possible Side Effects**
Potential Break in balance of overall railcraft, or at least the intended balance of liquid-fueled Steam boilers

*Further Work to be done*
Further balancing, testing/changing other blocks for interactions with these fluids

**Alternatives**

Different Balance ratio that is far weaker, based on comparing creosote to gaseous fuel. 
I chose not to use this method of balancing since
* Gaseous fuel is a lot more valuable than creosote (even keeping in mind the 'volume' difference)
* A generic refining setup would be incredibly inefficient at powering a boiler with these values
```java
// Ratio based on Gaseous fuel == Creosote
Fluids.OIL_HEAVY.ifPresent(f -> FluidFuelManager.addFuel(f, (int) (25600 * RailcraftConfig.boilerFuelMultiplier())));
Fluids.OIL_DENSE.ifPresent(f -> FluidFuelManager.addFuel(f, (int) (38400 * RailcraftConfig.boilerFuelMultiplier())));
Fluids.OIL_DISTILLED.ifPresent(f -> FluidFuelManager.addFuel(f, (int) (12000 * RailcraftConfig.boilerFuelMultiplier())));
Fluids.FUEL_DENSE.ifPresent(f -> FluidFuelManager.addFuel(f, (int) (115200 * RailcraftConfig.boilerFuelMultiplier())));
Fluids.FUEL_MIXED_HEAVY.ifPresent(f -> FluidFuelManager.addFuel(f, (int) (30720 * RailcraftConfig.boilerFuelMultiplier())));
Fluids.FUEL_LIGHT.ifPresent(f -> FluidFuelManager.addFuel(f, (int) (28800 * RailcraftConfig.boilerFuelMultiplier())));
Fluids.FUEL_MIXED_LIGHT.ifPresent(f -> FluidFuelManager.addFuel(f, (int) (9600 * RailcraftConfig.boilerFuelMultiplier())));
Fluids.FUEL_GASEOUS.ifPresent(f -> FluidFuelManager.addFuel(f, (int) (4800 * RailcraftConfig.boilerFuelMultiplier())));
```

### CODE REVIEW

**Review by liach — APPROVED** [review]
preferably you should add new enum entries after the,old values to preserve ordinal, but sure

**Comment by jasperfirecai2** [comment]
Like this?

File: src/main/java/mods/railcraft/common/fluids/Fluids.java [hunk 1]
```diff
@@ -30,7 +30,8 @@
  */
 public enum Fluids implements Supplier<@Nullable Fluid> {
 
-    WATER, LAVA, FUEL, BIOFUEL, IC2BIOGAS, CREOSOTE, STEAM, BIOETHANOL("bio.ethanol"), COAL, PYROTHEUM, FRESHWATER, BIODIESEL, DIESEL, GASOLINE;
+    FUEL, WATER, LAVA, BIOFUEL, IC2BIOGAS, CREOSOTE, STEAM, BIOETHANOL("bio.ethanol"), COAL, PYROTHEUM, FRESHWATER, BIODIESEL, DIESEL, GASOLINE,
```
Comment (liach) [inline src/main/java/mods/railcraft/common/fluids/Fluids.java]:
```suggestion
    WATER, LAVA, FUEL, BIOFUEL, IC2BIOGAS, CREOSOTE, STEAM, BIOETHANOL("bio.ethanol"), COAL, PYROTHEUM, FRESHWATER, BIODIESEL, DIESEL, GASOLINE,
```

**Review by liach — APPROVED** [review]

**Comment by jasperfirecai2** [comment]
Ah, right. my bad. Thanks!

**Comment by SHaaD94** [comment]
Hello guys. Could you please merge this PR?

**Pull request merged**